packages/xo-web/src/xo-app/host/tab-stats.js

```diff
@@ -4,56 +4,64 @@ import Icon from 'icon'
 import React from 'react'
 import Tooltip from 'tooltip'
 import { Container, Row, Col } from 'grid'
-import { DEFAULT_GRANULARITY, fetchStats, SelectGranularity } from 'stats'
+import { DEFAULT_GRANULARITY, fetchStats, SelectGranularity, INTERVAL_BY_GRANULARITY } from 'stats'
 import { Toggle } from 'form'
 import { CpuLineChart, MemoryLineChart, PifLineChart, LoadLineChart } from 'xo-line-chart'
 
 export default class HostStats extends Component {
   state = {
     granularity: DEFAULT_GRANULARITY,
     useCombinedValues: false,
+    statsIsPending: false,
   }
 
-  loop(host = this.props.host) {
+  fetchHostStats = () => {
+    if (this.state.statsIsPending) {
+      return
+    }
+
+    const host = this.props.host
+
     if (this.cancel) {
```

Comment:
When is this assigned a value?

Reply:
You're right, `this.cancel` is no longer assigned, that part of code is no longer needed either. I am going to remove it.